src/Microsoft.SqlTools.ServiceLayer/DisasterRecovery/DisasterRecoveryService.cs

```diff
@@ -11,14 +11,17 @@
 using Microsoft.SqlTools.ServiceLayer.Connection;
 using Microsoft.SqlTools.ServiceLayer.DisasterRecovery.Contracts;
 using Microsoft.SqlTools.ServiceLayer.Hosting;
+using Microsoft.SqlTools.ServiceLayer.TaskServices;
+using System.Threading;
 
 namespace Microsoft.SqlTools.ServiceLayer.DisasterRecovery
 {
     public class DisasterRecoveryService
     {
         private static readonly Lazy<DisasterRecoveryService> instance = new Lazy<DisasterRecoveryService>(() => new DisasterRecoveryService());
         private static ConnectionService connectionService = null;
-        private BackupUtilities backupUtilities;
+        private IBackupUtilities backupUtilities;
+        private ManualResetEvent manualResetEvent = new ManualResetEvent(initialState: false);
```

Comment:
>manualResetEvent [](start = 33, length = 16)

probably a better name would be something like "taskCompletedEvent" or something more specific about what the event is